File: examples/maven-spring-compile/src/main/java/community/flock/wirespec/example/maven/custom/app/todo/TodoRepository.java
package community.flock.wirespec.example.maven.custom.app.todo;

import community.flock.wirespec.generated.java.model.Todo;
import community.flock.wirespec.generated.java.model.TodoId;

import java.util.List;

public interface TodoRepository {
    List<Todo> getAllTodos();

    Todo getTodoById(final TodoId id);

    Todo saveTodo(final Todo todo);

    Todo deleteTodoById(final TodoId id);
}

File: examples/gradle-ktor/src/main/kotlin/community/flock/wirespec/example/maven/custom/app/todo/TodoRepository.kt
package community.flock.wirespec.example.maven.custom.app.todo

interface HasTodoRepository {
    val todoRepository: TodoRepository
}

interface TodoRepository {
    fun getAllTodos(): List<Todo>

    fun getTodoById(id: Todo.Id): Todo

    fun saveTodo(todo: Todo): Todo

    fun deleteTodoById(id: Todo.Id): Todo
}
